src/Aspire.Dashboard/Components/Pages/TraceDetail.razor.cs

```diff
@@ -115,8 +116,15 @@ private static Icon GetSpanIcon(OtlpSpan span)
 
     protected override async Task OnParametersSetAsync()
     {
-        UpdateDetailViewData();
-        UpdateSubscription();
+        if (TraceId != _trace?.TraceId)
```

Comment:
string comparer?

Reply:
We don't use one for traces anywhere else. TraceId is standard ordinal comparision, so equals and not equals is fine.

If it is changed to use an explicit comparison then it should be done across the app in a separate PR.